wrench/src/yaml_frame_writer.rs

```diff
@@ -324,40 +324,50 @@ impl YamlFrameWriter {
         let bytes = data.bytes.take().unwrap();
         let (path_file, path) = Self::next_rsrc_paths(&self.rsrc_prefix, &mut self.next_rsrc_num, &self.rsrc_base, "img", "png");
 
-        let ok = match data.format {
+        // takes a buffer with a stride and copies it into a new buffer that has stride == width
+        fn unstride<T: Clone>(mut slice: &[T], width: usize, stride: usize) -> Vec<T> {
+            let mut result = Vec::new();
+            while slice.len() > width {
+                result.extend_from_slice(&slice[..width]);
+                slice = &slice[stride..];
+            }
+            result.extend_from_slice(slice);
+            result
+        }
+
+        assert!(data.stride > 0);
+        match data.format {
```

Comment:
the body of the arms can be merged outside of the `match`, all it needs to return is `(ColorType, BytesPerTexel)` for the subsequent saving/unstriding